docs/docs/usage/hooks.md

```diff
@@ -0,0 +1,249 @@
+# Lifecycle and hooks
+
+As any Python deliverable, your project will go through the different phases
+of a Python project lifecycle and PDM provides commands to perform the expected tasks for those phases.
+
+It also provides hooks attached to these steps allowing for:
+
+- plugins to listen to the [signals][pdm.signals] of the same name.
```

Comment:
Is this a link ref or something? I can't find it

Reply:
More info: https://mkdocstrings.github.io/usage/#cross-references-to-any-markdown-heading (and previous/next paragraphs).

Reply:
That is awesome.